src/MiddlewareListener.php

```diff
@@ -69,7 +80,13 @@ public function onDispatch(MvcEvent $event)
             foreach ($routeMatch->getParams() as $key => $value) {
                 $psr7Request = $psr7Request->withAttribute($key, $value);
             }
-            $return = $middleware($psr7Request, Psr7Response::fromZend($response));
+            $return = $pipe(
```

Comment:
@asgrim So sorry I didn't notice this earlier, but... If we're pinning to Stratigility 2, we should likely use the http-interop `process()` method here instead of `__invoke()`, which means using a `DelegateInterface` instead of a callable.

May I suggest the following:

```php
$pipe->setResponsePrototype($psr7ResponsePrototype);
$return = $pipe->process($psr7Request, new \Zend\Stratigility\Delegate\CallableDelegateDecorator(
    function (PsrServerRequestInterface $request, PsrResponseInterface $response) {
        throw ReachedFinalHandlerException::create();
    }
));
```

That would accomplish the same as what you have here, and be forwards compatible.